functional-test/src/test/groovy/org/rundeck/util/container/BaseContainer.groovy

```
@@ -518,4 +522,48 @@ abstract class BaseContainer extends Specification implements ClientProvider, Wa
         response.successful ? MAPPER.readValue(response.body().string(), valueType) :  unsuccessfulResponseHandler(response)
     }
 
+
+    void dockerNodeAction(String action, String containerId) {
+        def process = "docker ${action} ${containerId}".execute()
+        def code = process.waitFor()
+        if (code != 0) {
+            println(process.getText())
+            throw new RuntimeException("Failed to ${action} container ${containerId}")
+        }
+    }
+
+    String getRundeckContainerId(){
+        if(RUNDECK_COMPOSE_CONTAINER){
+            return RUNDECK_COMPOSE_CONTAINER.getRundeckContainerId()
+        }
+        return RUNDECK_CONTAINER.containerId
+    }
+
+    void waitForRundeckUp(RdClient client){
+        int count = 0
+        def response = null
+        try {
+            response = client.doGet("/system/info")
+        } catch (InvokerInvocationException) {
+            //Here we just ignore the exception since i couldn't find the root cause
```

In general, swallowing exceptions is a bad practice.
You should `log` them with the appropriate level.
Otherwise, how would we ever know why the `"Rundeck did not start"`

---

This was updated with the wait and retry logic, I tried using the log var which gets defined by the @Slf4j annotation, but due to some lifecycle issues, considering we are restarting containers with a direct process instead of using TestContainers( which does not provide methods for this ) it caused a weird Exception. 

Now I log them by instantiating a logger directly in the catch clause. Also tried the List<Exception> accumulator and logging them after the waitfor, but the issue still popped up.